### ISSUE TITLE
Which model generated the answer?

### ISSUE THREAD

**Issue by fOO223Fr** [huggingface/text-generation-inference#1218]
### Feature request

TL;DR: /generate endpoint doesn't inform which model generated the text.

When there are multiple TGIS servers initiated with different models and a load balancer is placed infront of them then it becomes difficult to understand which model generated the text.

Below is a usecase where a gradio app is designed to gather feedback from users about the answer they got for their question.
![Untitled](https://github.com/huggingface/text-generation-inference/assets/10723238/18270e67-0302-4707-86d1-8bf06b37d828)

Note: i am aware of the /info endpoint which can give me the model_id but it doesnt help when multiple TGIS are behind a loadbalance because the same instance/server which got the /generate request may not get the /info request.

### Motivation

The motivation behind this is to gather feedback from users about different model's reply and eventually retrain accordingly.

### Your contribution

Yes i can help. i have already modified the router code and the text_generation client code to facilitate my usecase and i will create a PR with the same.

**Comment by OlivierDehaene** [comment]
Why is the load balancer not adding a header to track the request instead?

**Comment by fOO223Fr** [comment]
if i understand correctly the LB should set a Header like `X-Backend-Server` to a request/response. Consequently, then the app should parse the header and if it is from server-1 it would determine it is from model-1, is that a correct understanding?

**Comment by fOO223Fr** [comment]
if that is what you mean then problem is the app has to have the **hard coded** knowledge of the relationship between which server has which model populated. Which can help in my application surely but having hard coded knowledge is not very appreciated therefore, i still feel the /generate and consequently, /generate_stream should give users an important information which is "Which model generated the answer?"
